tests/core/providers/test_websocket_provider.py

```diff
@@ -8,6 +21,40 @@
 )
 
 
+@pytest.yield_fixture
+def start_websocket_server(open_port):
+    event_loop = asyncio.new_event_loop()
+
+    def run_server():
+        async def empty_server(websocket, path):
+            data = await websocket.recv()
+            await asyncio.sleep(0.1)
+            await websocket.send(data)
+        server = websockets.serve(empty_server, '127.0.0.1', open_port, loop=event_loop)
+        event_loop.run_until_complete(server)
+        event_loop.run_forever()
+
+    thd = Thread(target=run_server)
+    thd.start()
+    yield
+    event_loop.call_soon_threadsafe(event_loop.stop)
+
+
+@pytest.fixture()
+def w3(open_port, start_websocket_server):
+    # need new event loop as the one used by server is already running
+    event_loop = asyncio.new_event_loop()
+    endpoint_uri = 'ws://127.0.0.1:{}'.format(open_port)
+    event_loop.run_until_complete(wait_for_ws(endpoint_uri, event_loop))
+    provider = WebsocketProvider(endpoint_uri, websocket_timeout=0)
```

Comment:
I'm inclined to suggest changing this timout to be `0.01` since `0` is fundamentally different.  While it's possible this will work fine, I could see a timeout of `0` having some fundamentally different behavior (like unconditionally raising a timeout even if the response was instance) than a very small one.